libraries/communication/rtClient/rtcmdclient.h

```diff
@@ -199,6 +199,7 @@ class COMMUNICATIONSHARED_EXPORT RtCmdClient : public QTcpSocket
     CommandManager  m_commandManager;   /**< The command manager. */
     QMutex          m_qMutex;           /**< Access serialization between threads */
     QString         m_sAvailableData;   /**< The last received response. */
+    using QTcpSocket::connectToHost;
```

Comment:
why is this needed?

Reply:
becuase the signature is different, and the compiler warns you that you are actually not overriding.

Reply:
Can you share the exact warning please and also where it is shown? I assume https://github.com/mne-tools/mne-cpp/blob/98f9741f82d70fd489882ad2d73f8af0eeee5678/libraries/communication/rtClient/rtcmdclient.cpp#L75

Reply:
![image](https://user-images.githubusercontent.com/8955424/78398803-56918b80-75b9-11ea-8aed-dbbae2e1e262.png)

Reply:
thx. Now I understand. so what happens is that the name of this function was chosen poorly since it is identical with the one in QAbstractSocket and thus tries to overload it. Honestly, I do not understand how `using QTcpSocket::connectToHost;` gets rid of the warning, but it does 

We try not to use `using` statements in headers. This [article](https://stackoverflow.com/questions/5849457/using-namespace-in-c-headers) explains why that is.

How about if we overload the function properly?

```
//=========================================================================================================
/**
 * Connect to a mne_rt_server using port 4217
 *
 * @param[in] hostName      The hostName on the given port.
 * @param[in] port          The port.
 * @param[in] openMode      The open mode. Default is QIODevice::OpenMode::ReadWrite.
 * @param[in] protocol      The protocol. Default is QAbstractSocket::NetworkLayerProtocol::AnyIPProtocol.
 */
void connectToHost(const QString &hostName,
                   quint16 port,
                   QIODevice::OpenMode openMode = ReadWrite,
                   QAbstractSocket::NetworkLayerProtocol protocol = AnyIPProtocol);
```

```
//=============================================================================================================

void RtCmdClient::connectToHost(const QString &hostName,
                                quint16 port,
                                QIODevice::OpenMode openMode,
                                QAbstractSocket::NetworkLayerProtocol protocol)
{
    Q_UNUSED(port)
    QTcpSocket::connectToHost(hostName, 4217, openMode, protocol);
}
```

Reply:
What about this call?
https://github.com/mne-tools/mne-cpp/blob/eb75b90aa61a9ccedbfd2cfaed82ff38b330284f/applications/mne_scan/plugins/fiffsimulator/fiffsimulator.cpp#L285

I don't really like: ``` m_pRtCmdClient->connectToHost(m_sFiffSimulatorIP,0);  ```

If we define a pointer to a qthread object, we can call its method, and our lives would be much happier.

Reply:
> define a pointer to a qthread object, we can call its method

What do you mean by that exactly? 

> I don't really like: m_pRtCmdClient->connectToHost(m_sFiffSimulatorIP,0);

Where does the second parameter (`0`) come from?

How about we set the `port ` to default to 4217? This not also gets rif of the warning but also opens up the possibility to connect to a different port sinc ethe port parameter is now actually use.

```
//=========================================================================================================
/**
 * Connect to a mne_rt_server using port 4217 as default.
 *
 * @param[in] hostName      The hostName on the given port.
 * @param[in] port          The port. Default is 4217.
 * @param[in] openMode      The open mode. Default is QIODevice::OpenMode::ReadWrite.
 * @param[in] protocol      The protocol. Default is QAbstractSocket::NetworkLayerProtocol::AnyIPProtocol.
 */
void connectToHost(const QString &hostName,
                   quint16 port = 4217,
                   QIODevice::OpenMode openMode = ReadWrite,
                   QAbstractSocket::NetworkLayerProtocol protocol = AnyIPProtocol);
```

```
//=============================================================================================================

void RtCmdClient::connectToHost(const QString &hostName,
                                quint16 port,
                                QIODevice::OpenMode openMode,
                                QAbstractSocket::NetworkLayerProtocol protocol)
{
    QTcpSocket::connectToHost(hostName, port, openMode, protocol);
}
```

Reply:
OK, even better. I like this so much more. I thought you wanted to keep the ingrained port number.